tests/tests_lite/plugins/collectives/test_torch_.py

```diff
@@ -226,18 +234,21 @@ def test_collectives_distributed(n):
 
 
 def _test_two_groups(strategy, left_collective, right_collective):
-    left_collective.create_group(ranks=[0, 1])
-    right_collective.create_group(ranks=[1, 2])
+    with check_destroy_group():  # the fixture does not work inside the launched function, use it manually
+        left_collective.create_group(ranks=[0, 1])
+        right_collective.create_group(ranks=[1, 2])
 
-    if strategy.global_rank in (0, 1):
-        tensor = torch.tensor(strategy.global_rank)
-        left_collective.all_reduce(tensor)
-        assert tensor == 1
-    right_collective.barrier()
-    if right_collective.rank >= 0:
```

Comment:
the problem was also this line, which should have been `rank in (1, 2)` as per my original demo script :)